wpilibj/src/shared/java/edu/wpi/first/wpilibj/command/Scheduler.java

```diff
@@ -274,9 +237,8 @@ void remove(Command command) {
    */
   public void removeAll() {
     // TODO: Confirm that this works with "uninteruptible" commands
-    while (m_firstCommand != null) {
-      remove(m_firstCommand.getData());
-    }
+    // Use a temporary view to avoid concurrent modification
+    new ArrayList<>(m_commands).forEach(this::remove);
```

Comment:
Because the scheduler calls `remove` on each command. This would reverse the order that that's called.